File: src/main/java/com/example/lms/service/LeaveTypeService.java
package com.example.lms.service;

import com.example.lms.entity.LeaveType;
import com.example.lms.repository.LeaveTypeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface LeaveTypeService {
    List<LeaveType> getAllLeaveTypes();

    Optional<LeaveType> getLeaveTypeById(int typeId);

    LeaveType saveLeaveType(LeaveType leaveType);

    void deleteLeaveType(int typeId);

    LeaveType updateLeaveType(int leaveTypeId, LeaveType leaveType);
}

File: leavemanagement1ui/src/services/LeaveTypeService.js
import axios from 'axios';

const LEAVE_TYPE_API_BASE_URL = 'http://localhost:8089/api/leave-types';

class LeaveTypeService {
  getAllLeaveTypes() {
    return axios.get(LEAVE_TYPE_API_BASE_URL);
  }

  getLeaveTypeById(leaveTypeId) {
    return axios.get(`${LEAVE_TYPE_API_BASE_URL}/${leaveTypeId}`);
  }

  saveLeaveType(leaveType) {
    return axios.post(LEAVE_TYPE_API_BASE_URL, leaveType);
  }

  updateLeaveType(leaveTypeId, leaveType) {
    return axios.put(`${LEAVE_TYPE_API_BASE_URL}/${leaveTypeId}`, leaveType);
  }

  deleteLeaveType(leaveTypeId) {
    return axios.delete(`${LEAVE_TYPE_API_BASE_URL}/${leaveTypeId}`);
  }
}

export default new LeaveTypeService();
